### ISSUE TITLE
Custom resolver value not used for sorting in page query

### ISSUE THREAD

**Issue by jabrks** [gatsbyjs/gatsby#27368]
## Description

Introduced in `gatsby@2.24.51` as a result of [#26644](https://github.com/gatsbyjs/gatsby/pull/26644).

I started to look into what the cause of the issue might be and I've shared my findings below.

We have a custom resolver for a nested Date field, and a static query with a sort on that Date field. We also have a page query with both a sort on the Date field and a filter on a Category field.

1. When running the static query, Gatsby determines that the Date subfield of the Meta field needs to be resolved before sorting and resolves it.

2. When running the page query, Gatsby knows that Date has already been resolved in the static query so it doesn't need to be resolved again, but Categories is a new nested field that needs to be resolved. https://github.com/gatsbyjs/gatsby/blob/7b37bb7d6aa09bc2f1eb19df0a23d2b909425f56/packages/gatsby/src/schema/node-model.js#L369-L372

3. Gatsby resolves the Categories field, and the value for Date is populated from the same node (which doesn't have the value of Date from the custom resolver).
https://github.com/gatsbyjs/gatsby/blob/7b37bb7d6aa09bc2f1eb19df0a23d2b909425f56/packages/gatsby/src/schema/node-model.js#L711-L715

4. The query results are sorted using the original value of the Date field instead of the value generated by the custom resolver and they're returned in the wrong order.

### Steps to reproduce

<https://codesandbox.io/s/gatsby-custom-resolver-bug-fy7zn>

### Expected result

The query results should be sorted using the value of Date generated by the custom resolver.

### Actual result

The query results are sorted using the original value of Date.

### Environment

```bash
System:
  OS: macOS 10.15.7
  CPU: (8) x64 Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
  Shell: 5.7.1 - /bin/zsh
Binaries:
  Node: 12.18.4 - ~/.nvm/versions/node/v12.18.4/bin/node
  Yarn: 1.22.5 - /usr/local/bin/yarn
  npm: 6.14.8 - ~/.nvm/versions/node/v12.18.4/bin/npm
Languages:
  Python: 2.7.17 - /usr/local/bin/python
Browsers:
  Chrome: 86.0.4240.75
  Edge: 86.0.622.38
  Firefox: 81.0.1
  Safari: 14.0
npmPackages:
  gatsby: ^2.24.73 => 2.24.73 
  gatsby-image: ^2.4.20 => 2.4.20 
  gatsby-plugin-amp: ^0.2.2 => 0.2.2 
  gatsby-plugin-feed: ^2.5.13 => 2.5.13 
  gatsby-plugin-google-tagmanager: ^2.3.13 => 2.3.13 
  gatsby-plugin-react-helmet: ^3.3.12 => 3.3.12 
  gatsby-plugin-remove-generator: ^1.0.6 => 1.0.6 
  gatsby-plugin-sass: ^2.3.14 => 2.3.14 
  gatsby-plugin-sharp: ^2.6.37 => 2.6.38 
  gatsby-plugin-sitemap: ^2.4.14 => 2.4.14 
  gatsby-plugin-twitter: ^2.3.12 => 2.3.12 
  gatsby-remark-copy-linked-files: ^2.3.17 => 2.3.17 
  gatsby-source-filesystem: ^2.3.31 => 2.3.31 
  gatsby-transformer-remark: ^2.8.36 => 2.8.36 
  gatsby-transformer-sharp: ^2.5.16 => 2.5.16 
  gatsby-transformer-yaml: ^2.4.13 => 2.4.13 
npmGlobalPackages:
  gatsby-cli: 2.12.106
  gatsby-dev-cli: 2.7.21
```

**Comment by vladar** [comment]
Hey @jabrks !

Thanks for opening this and for a thorough investigation! 💜 

Should be fixed in #27411 Let me know if it works for you if you have a chance.

**Comment by jabrks** [comment]
Thanks for the quick fix @vladar!

I've just checked out your branch locally and the issue seems to be resolved.

**Comment by vladar** [comment]
Published in `gatsby@2.24.76`
